sourcemap/elasticsearch.go

```diff
@@ -0,0 +1,115 @@
+package sourcemap
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-sourcemap/sourcemap"
+
+	"github.com/elastic/beats/libbeat/common"
+	es "github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type elasticsearch interface {
+	fetch(id Id) (*sourcemap.Consumer, error)
+}
+
+type smapElasticsearch struct {
+	clients []es.Client
+	index   string
+}
+
+func NewElasticsearch(config *common.Config, index string) (*smapElasticsearch, error) {
+	esClients, err := es.NewElasticsearchClients(config)
+	if err != nil || esClients == nil || len(esClients) == 0 {
```

Comment:
looking at es.NewElasticsearchClients, part of this condition is redundant, `esClients == nil || len(esClients) == 0` can not happen here.
shall i create a pr for this?